components/merkledb/examples/migration/mod.rs

```diff
@@ -181,109 +181,47 @@ mod v2 {
-            // and don't migrate the wallet.
-            new_data.create_tombstone(("histories", &public_key));
-        } else {
-            // Merkelize the wallet history.
-            let mut history = new_schema.histories.get(&public_key);
-            history.extend(&old_schema.histories.get(&public_key));
-
-            let new_wallet = v2::Wallet {
-                username: wallet.username,
-                balance: wallet.balance,
-                history_hash: history.object_hash(),
-            };
-            new_schema.wallets.put(&public_key, new_wallet);
-        }
-
-        if i % 1_000 == 999 {
-            println!("Processed {} wallets", i + 1);
-        }
-    }
-}
-
-fn main() {
-    let db = TemporaryDB::new();
-    let fork = db.fork();
-    create_initial_data(&fork);
-    fork.get_proof_list("unrelated.list").extend(vec![1, 2, 3]);
-    db.merge(fork.into_patch()).unwrap();
-
-    let fork = db.fork();
-    let new_data = Migration::new("test", &fork);
-    let old_data = Prefixed::new("test", fork.readonly());
-    {
-        let old_schema = v1::Schema::new(old_data.clone());
-        println!("Before migration:");
-        old_schema.print_wallets();
-    }
-    create_migration(new_data, old_data);
-    db.merge(fork.into_patch()).unwrap();
-
-    // For now, the old data is still present in the storage.
-    let snapshot = db.snapshot();
-    let old_schema = v1::Schema::new(Prefixed::new("test", &snapshot));
+/// Checks that we have old and new data in the storage after migration.
+pub fn check_data_before_flush(snapshot: &dyn Snapshot) {
+    let old_schema = v1::Schema::new(Prefixed::new("test", snapshot));
     assert_eq!(old_schema.ticker.get().unwrap(), "XNM");
     // The new data is present, too, in the unmerged form.
-    let new_schema = v2::Schema::new(Migration::new("test", &snapshot));
+    let new_schema = v2::Schema::new(Migration::new("test", snapshot));
     assert_eq!(new_schema.config.get().unwrap().ticker, "XNM");
 
-    let system_schema = SystemSchema::new(&snapshot);
+    let system_schema = SystemSchema::new(snapshot);
     let state = system_schema.state_aggregator();
     assert_eq!(state.keys().collect::<Vec<_>>(), vec!["unrelated.list"]);
-    let migration_view = Migration::new("test", &snapshot);
+    let migration_view = Migration::new("test", snapshot);
     let state = migration_view.state_aggregator();
     assert_eq!(
         state.keys().collect::<Vec<_>>(),
         vec!["test.config", "test.wallets"]
     );
     let new_state_hash = state.object_hash();
     assert_eq!(new_state_hash, migration_view.state_hash());
+}
 
-    let mut fork = db.fork();
-    flush_migration(&mut fork, "test");
-    let patch = fork.into_patch();
-
+/// Checks that old data was replaced by new data in the patch
+/// after we called `flush_migration`.
+pub fn check_data_after_flush(patch: &Patch) {
     // Now, the new indexes have replaced the old ones.
```

Comment:
Nit: This comment looks redundant given that the method purpose is commented right above.